ee/models/event_definition.py

```diff
@@ -0,0 +1,12 @@
+from django.contrib.postgres.fields import ArrayField
+from django.db import models
+
+from posthog.models.event_definition import EventDefinition
+
+
+class EnterpriseEventDefinition(EventDefinition):
+    owner = models.ForeignKey("posthog.User", null=True, on_delete=models.PROTECT, related_name="event_definitions",)
```

Comment:
Probably should be `models.SET_NULL`. Don't think this should block user deletion.

Reply:
I thought this would protect the event definition from being deleted if a user is deleted? Is owner not the parent here? I might have it mixed up?

Reply:
Does setting the on_delete to models.SET_NULL mean that if a user is deleted, we're still keeping the event_definition, just setting its "owner" field to null?

Reply:
So with `models.PROTECT` the `User` object will not be allowed to be deleted while they have a related `EnterpriseEventDefinition` object. With `SET_NULL` exactly that, the object is kept and we just set the FK to `null`. Reference: https://docs.djangoproject.com/en/3.2/ref/models/fields/#django.db.models.SET_NULL